### PR TITLE
[Impeller] fix subpass collapse pass ending logic

### PR DESCRIPTION
**Pull request by jonahwilliams**
Discovered during investigation of opacity optimizations, currently OnRender will unconditionally end the current pass - but this is incorrect if we're merged into the parent pass. Detect this case and ensure that the collapsed child gets the same render target and render pass.

### CODE REVIEW

File: impeller/entity/entity_unittests.cc [hunk 1]
```diff
@@ -85,15 +85,19 @@ class TestPassDelegate final : public EntityPassDelegate {
 
  private:
   const std::optional<Rect> coverage_;
+  bool collapse_;
```
Comment (chinmaygarde) [inline impeller/entity/entity_unittests.cc:88]: Nit: Default it, or if it set in a ctor, `const bool`

**Review by chinmaygarde — APPROVED** [review]

File: impeller/entity/entity_unittests.cc (hunk 1 above)
Comment (jonahwilliams) [reply impeller/entity/entity_unittests.cc]: Done

File: impeller/entity/inline_pass_context.h [hunk 2]
```diff
@@ -38,6 +40,7 @@ class InlinePassContext {
   std::shared_ptr<RenderPass> pass_;
   uint32_t pass_count_ = 0;
   uint32_t total_pass_reads_ = 0;
+  bool collapsed_parent_pass_ = false;
```
Comment (bdero) [inline impeller/entity/inline_pass_context.h:43]: nit: `is_collapsed_`?

**Review by bdero — APPROVED** [review]

File: impeller/entity/inline_pass_context.h (hunk 2 above)
Comment (jonahwilliams) [reply impeller/entity/inline_pass_context.h]: Done

**Pull request merged**